### PR TITLE
Mark `go_sdk` module extension as `{os,arch}_dependent`

### PR DESCRIPTION
**Pull request by fmeum**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

This ensures that the lockfile maintains separate instances of this extension for each OS/arch combination.

### CODE REVIEW

**Comment by fmeum** [comment]
CC @tyler-french

File: go/private/extensions.bzl [hunk 1]
```diff
@@ -207,12 +208,21 @@ def _left_pad_zero(index, length):
         fail("index must be non-negative")
     return ("0" * length + str(index))[-length:]
 
+go_sdk_extra_kwargs = {
+    # The choice of a host-compatible SDK is expressed in repository rule attribute values and
+    # depends on host OS and architecture.
+    "os_dependent": True,
+    "arch_dependent": True,
+} if bazel_features.external_deps.module_extension_has_os_arch_dependent else {}
+print(go_sdk_extra_kwargs)
```
Comment (linzhp) [inline go/private/extensions.bzl:217]: debug statement :D

Comment (fmeum) [reply go/private/extensions.bzl:217]: Whoops :-)

Comment (fmeum) [inline go/private/extensions.bzl:217]:
```suggestion
```

**Review by linzhp — APPROVED** [review]

**Comment by fmeum** [comment]
@tpasternak There isn't and given that a module extension can run essentially arbitrary binaries, this can't be done. I'm looking into ways to resolve the situation for `go_sdk` and similar extensions.

**Pull request merged**